task/gitlab-set-status/0.1/gitlab-set-status.yaml

```diff
@@ -0,0 +1,140 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: gitlab-set-status
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/categories: Messaging
+    tekton.dev/tags: gitlab, git
+    tekton.dev/displayName: "Set Gitlab commit status"
+spec:
+  description: >-
+    This task will set the status of the CI job to the specified value along
+    with a link to the specified target URL where developers can follow the
+    progress of the CI job.
+
+    The `gitlab-set-status` task allows external services to mark GitLab commits
+    with an `error`, `failure`, `pending`, or `success` state, which is then
+    reflected in pull requests involving those commits. Statuses include as well a
+    `description` and a `target_url`, to give the user informations about the CI
+    statuses or a direct link to the full log.
+
+  params:
+    - name: GITLAB_HOST_URL
+      description: |
+        The GitLab host, adjust this if you run a GitLab enterprise.
+      default: "gitlab.com"
+      type: string
+
+    - name: API_PATH_PREFIX
+      description: |
+        The API path prefix, GitLab Enterprise has a prefix e.g. /api/v4
+      default: "/api/v4"
+      type: string
+
+    - name: REPO_FULL_NAME
+      description: |
+        The GitLab repository full name, e.g.: tektoncd/catalog
+      type: string
+
+    - name: GITLAB_TOKEN_SECRET_NAME
+      description: |
+        The name of the kubernetes secret that contains the GitLab token, default: gitlab-api-secret
+      type: string
+      default: gitlab-api-secret
+
+    - name: GITLAB_TOKEN_SECRET_KEY
+      description: |
+        The key within the kubernetes secret that contains the GitLab token, default: token
+      type: string
+      default: token
+
+    - name: SHA
+      description: |
+        Commit SHA to set the status for.
+      type: string
+
+    - name: TARGET_URL
+      description: |
+        The target URL to associate with this status. This URL will be linked
+        from the GitLab UI to allow users to easily see the source of the
+        status.
+      type: string
+
+    - name: DESCRIPTION
+      description: |
+        A short description of the status.
+      type: string
+
+    - name: CONTEXT
+      description: |
+        The GitLab context, A string label to differentiate this status from
+        the status of other systems. ie: "continuous-integration/tekton"
+      default: "continuous-integration/tekton"
+      type: string
+
+    - name: STATE
+      description: |
+        The state of the status. Can be one of the following `pending`,
+        `running`, `success`, `failed`, or `canceled`.
+      type: string
+
+  steps:
+    - name: set-status
+      image: broadinstitute/python-requests:latest@sha256:377ab24f9253f2d089fff932a1b79d4b3eefa520d7d33167a7f521b49d1415e3
```

Comment:
can we use the image `registry.access.redhat.com/ubi8/ubi-minimal:8.2` as it is a maintained one?

Reply:
I could, but then you don't have the `requests` library installed, which means you have to do flimsy params parsing for every gitlab task we add. I do agree that it is less than ideal to use an image like this, though. Too bad you can't share code between tasks..
If you want, I could rewrite the code around `http.client`, but it feels wrong when you have a library like `requests` at your disposal.

Reply:
I agree that we don't get `requests` library installed but the image was last updated 3years back and may have some critical CVEs. If possible can you please try looking for another image which is well maintained?

Reply:
I ended up using `http.client` as well, since there are no actively maintained `requests` images. It's suprising to me, because even the `http.client` docs point to `requests` as the better option. But okay, that's a topic for a different discussion.

Let me know if it's good like this, then I'll squash the commits.

Reply:
yeah looks good. Probably you can squash the commits and then we can merge this

Reply:
Done